shared/src/components/CodeExcerpt.test.tsx

```diff
@@ -104,4 +107,24 @@ describe('CodeExcerpt', () => {
             expect(span.textContent === 'test')
         }
     })
+
+    it('does not disable the highlighting timeout', () => {
+        /*
+            Because disabling the timeout should only ever be done in response
+            to the user asking us to do so, something that we do not do for
+            code excerpts because falling back to plaintext rendering is most
+            ideal.
+        */
+        const fetchHighlightedFileLines = sinon.fake.returns(of(HIGHLIGHTED_FILE_LINES))
```

Comment:
```suggestion
        const fetchHighlightedFileLines = sinon.spy(() => of(HIGHLIGHTED_FILE_LINES))
```
This is more type safe (the spy actually encodes the parameter and return types, the fake API can't do that afaik)